proxy_test.go

```diff
@@ -666,7 +687,19 @@ var (
 				if err != nil {
 					panic(fmt.Sprintf("error while imitating delay at fakeServer handler: %s", err))
 				}
-				time.Sleep(d)
+
+				// at this step the query is really processed by the server
+				// we're only interested by long queries, i.e the one taking more than 0 msec
+				if d == heavyRequestDuration {
+					atomic.AddInt64(&nbHeavyRequestsInflight, 1)
+				}
+
+				// instead of sleeping for the whole duration, we sleep msec per msec so that the query can be
+				// cancel once it's associated test is over because this test suite generate a lot of goroutines,
+				// which can be an issue on small cpu like github actions
+				for i := int64(0); i < d.Milliseconds() && atomic.LoadUint64(&shouldStop) == 0; i++ {
```

Comment:
This way we allow scheduler to assign desired go routine to a kernel thread for real execution?

Reply:
no, this loop is just here to cancel the query of a previous test.  When a new test starts, we put shouldStop at 1 then wait to avoid too many sleeping go-routines. I'm not sure if it's that much impactful since now we can have up to 256 concurrent sleeping goroutines but since the process running a github actions is so light in terms of CPU I thought it would be better to avoid useless concurrent goroutines